py/stabilizer/stream.py

```diff
@@ -4,11 +4,54 @@
 
 Description: Provides a means of accessing Stabilizer livestream data.
 """
-import socket
-import time
+import asyncio
+import collections
+import copy
 import logging
+import select
+import socket
 import struct
 
+from typing import Mapping, ByteString, List, Callable, Union
+
+import numpy as np
+
+from . import DAC_VOLTS_PER_LSB, ADC_VOLTS_PER_LSB
+
+StreamFrame = collections.namedtuple('StreamFrame', ['sequence_number', 'format', 'traces'])
+"""
+A single frame received over the livestream interface.
+
+# Args
+* `sequence_number` - The starting sample sequence number.
+* `format` - The format code of the frame.
+* `traces` - A dictionary of traces, each entry of which contain the raw data associated with the
```

Comment:
I don't think a generic frame payload will always be `traces`.

Reply:
The main thought here is that we're using the stream to visualize time-series data. What are some use cases where we don't anticipate the stream frame to have that type of data?

Reply:
I'm a bit torn about where the payload parsing should happen. I'll think about it.

Reply:
There's certainly something to be said about offloading the parsing of the frame data... The downside if we don't unify it is that we would potentially be copying code in various places. Alternatively, we could just provide some parsing functions and not explicitly parse the frame, and just provide the raw data that the end user can pass to the provided parsing functions. That would definitely simplify the `formatter` API.

Downside of that approach is that we have to carry around the concept of "batches" until the frame has been deserialized.